.github/workflows/spelling2.yml

```diff
@@ -12,21 +12,83 @@ on:
     tags-ignore:
       - "**"
     types: ['opened', 'reopened', 'synchronize']
+  issue_comment:
+    types: [created]
 
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+    outputs:
+      followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
+    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
     steps:
-    - name: checkout-merge
-      if: "contains(github.event_name, 'pull_request')"
-      uses: actions/checkout@v2
-      with:
-        ref: refs/pull/${{github.event.pull_request.number}}/merge
-    - name: checkout
-      if: ${{ github.event_name == 'push' }}
-      uses: actions/checkout@v2
-    - uses: check-spelling/check-spelling@v0.0.19
+    - name: check-spelling
       id: spelling
+      uses: check-spelling/check-spelling@v0.0.20
+      with:
+        config: .github/actions/spell-check
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'powerdns' && 1 || 0 }}
+        suppress_push_for_open_pull_request: 1
+        checkout: true
+        post_comment: 0
+        extra_dictionaries:
+          cspell:aws/aws.txt
+          cspell:filetypes/filetypes.txt
+          cspell:django/django.txt
+          cspell:html/html.txt
+          cspell:fullstack/fullstack.txt
+          cspell:npm/npm.txt
+          cspell:lua/lua.txt
+          cspell:python/python.txt
+          cspell:css/css.txt
+        check_extra_dictionaries: ''
+
+  comment:
+    name: Report
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
This enables commenting on a commit or PR respectively.

Reply:
does it also allow any other kind of write? to other PRs perhaps?

Reply:
what does `contents: write` enable?

Reply:
Sadly way too much (it includes the ability to push to the repository). It's here to enable a comment to be posted on `push`.
https://docs.github.com/en/rest/overview/permissions-required-for-github-apps#permission-on-contents

It's really unfortunate that github doesn't have a `comment` permission and instead lumps the ability to `comment` into either `contents: write` (for commenting on  commits) or `pull-requests: write` (for commenting on pull requests).

If this is a concern, this could be split into two different jobs.

One job could run for forks and include `contents: write`, and a different job could run for the primary repository and not include it.

The privilege isolation means that the code that's running should only be posting a comment based on the data generated by the job which takes untrusted input and checks its spelling. -- It should be possible for you to review the commenting codepath if you're concerned (and I'd welcome that review).

I'm going to investigate using [`GITHUB_STEP_SUMMARY`](
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary) instead of comments (in addition to trying to get sarif to work). I think it might be possible.

Also note that instead of using `@v0.0.20` I could change the commit to use the equivalent SHA if you're concerned that the tag could move -- I have no intention of ever moving a tag, and w/ this version if I need to disable a release, I can do it via [secpoll](https://github.com/check-spelling/check-spelling/wiki/Feature:-Security-Polling).